util/chunk/codec.go

```diff
@@ -189,3 +189,138 @@ func init() {
 		allNotNullBitmap[i] = 0xFF
 	}
 }
+
+// Decoder decodes the data returned from the coprocessor and stores the result in Chunk.
+// How Decoder works:
+// 1. Initialization phase: Decode a whole input byte slice to Decoder.intermChk using Codec.Decode. intermChk is
+//    introduced to simplify the implementation of decode phase. This phase uses pointer operations with less CPU andF
+//    memory cost.
+// 2. Decode phase:
+//    2.1 Set the number of rows that should be decoded to a multiple of 8 greater than
+//        chk.RequiredRows() - chk.NumRows(), this can reduce the cost when copying the srcCol.nullBitMap into destCol.nullBitMap.
```

Comment:
This line is confusing me. 
@yikeke PTAL.

Reply:
> This line is confusing me.
> @yikeke PTAL.

Working on it with @wshwsh12 .

Reply:
My suggestion: 

Set the number of rows to be decoded to a value that is a multiple of 8 and greater than `chk.RequiredRows() - chk.NumRows()`. This reduces the overhead of copying the srcCol.nullBitMap into destCol.nullBitMap.

Reply:
update.